src/v0/destinations/ga4/utils.js

```diff
@@ -554,21 +554,17 @@ const buildDeliverablePayload = (payload, Config) => {
   return response;
 };
 
-const sanitizeUserProperties = (userProperties) => {
-  Object.keys(userProperties).forEach((key) => {
-    const propetyValue = userProperties[key];
-    if (
-      typeof propetyValue === 'string' ||
-      typeof propetyValue === 'number' ||
-      typeof propetyValue === 'boolean'
-    ) {
-      delete userProperties[key];
-      userProperties[key] = {
-        value: propetyValue,
-      };
+function sanitizeUserProperties(obj) {
```

Comment:
you don't want to mutate userProperties is that the change?

Reply:
No its basically I want to sanitize for any properties inside userProperties which are non primitive values